City-of-Helsinki/berth-reservations-ui
# VEN-1230, VEN-1238 | Replace the API with federation API

## Description :sparkles:

* Replaced the API with federation API to allow removing the legacy berths app
  * Refactored some components to use the term `harbor` if they relate to harbors instead of berths, for clarity
* Replaced harbor, pier and berth inputs on berth switch application form
  * Refactored related components and structure in the process
  * Updated HDS to 1.0.0
* Code related to VEN-1312 is commented out until it's implemented

## Issues :bug:

### Closes :no_good_woman:

* [VEN-1230](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1230): FE - Use Federation on Customer UI
* [VEN-1238](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1238): FE - Replace Customer UI pier and berth number inputs

## Testing :alembic:

### Automated tests :gear:️

* Updated related tests

### Manual testing :construction_worker_man:

* Creating one of each application type should work as before
* Switch applications should now require selecting harbor, pier and berth
